### DISCUSSION TITLE
AUTOPARSE feature

### DISCUSSION THREAD

**Discussion by johnbent** [tekinged/missing#348] [Show and tell]
When an Issue is ready to be imported into the database, two things need to be done:
1. A comment should be added that has 'AUTOPARSE' as its first line and then special formatting that the parser can parse
  - Follow the instructions here: https://github.com/tekinged/tekinged.com/blob/03b3b66fa10876d48dba9e6e39c19c2a4e93c231/scripts/add_words.py#L17
2. A label of 'ready for import' should be added.

Examples of an Issue that was successfully imported using this automation can be seen here: https://github.com/tekinged/missing/issues?q=label%3ANOT_IMPORTED+is%3Aclosed

**Comment by johnbent** [comment]
To add a label, when you are viewing an Issue, click on the settings wheel next to the Labels section:
![image](https://user-images.githubusercontent.com/2047294/196489465-1ba315f3-5059-4b29-b919-e7b54456290e.png)

In the image above, you can see the Labels section and its settings wheel in the bottom right. After you click on it, then select 'ready for import':

![image](https://user-images.githubusercontent.com/2047294/196489681-ce377b6a-70e9-45c2-b05d-1103e49bf74d.png)

Then let me know that it is ready and I'll run the script to do the auto-import and close the issue!

**Reply by ChrisPerrette** [reply]
I guess I don't have permission to do this. The settings wheel doesn't show up for me.

**Reply by johnbent** [reply]
Aha. Thanks for testing. You did not have Write permissions on the repo. I just created a Team and invited you into it. You should have an invite email from github. Please accept that invite and then try again. Thanks for the patience; just a one time thing!

**Reply by ChrisPerrette** [reply]
It is working now, thanks.

**Comment by smith-371** [comment]
the problem is I don't see the settings wheel

**Comment by johnbent** [comment]
Check your spam folder for the email you use for github. You will have received an invitation from GitHub to join the tekinged Organization and the Committee Team. Accept that invite and you'll be good.

**Comment by johnbent** [comment]
![image](https://user-images.githubusercontent.com/2047294/196534636-3b196f46-a687-416f-944c-73b1b975f67b.png)
Your invitation is "pending". Once you accept, you'll have the settings wheel.

**Comment by smith-371** [comment]
AUTOPARSE
l cheroll ulemcheroll
l delebaob melebaob

**Reply by johnbent** [reply]
It won't work here. This looks great but please do the following:
1. Create a new Issue called "Words Needing Linking"
2. Copy this into a comment in that new Issue
3. Add the "ready for import" label

**Comment by smith-371** [comment]
does it work correctly now?

**Comment by johnbent** [comment]
@smith-371 , do you remember how we passed expressions to the parser?

**Reply by johnbent** [reply]
I'm just gonna use underscores for now and then deal with them later. I'm not sure how it used to work.

**Comment by smith-371** [comment]
try this: 

w chur n. xx;
p mengemkim er a cheril --holding back laughter, stifle laughter, hide laughter

**Reply by johnbent** [reply]
Awesome; thanks!

**Comment by johnbent** [comment]
By the way, be careful please about saying autoparse in a comment. As lower-case it is fine, but only use the uppercase version when you are ready for the parser. If you just discuss it, use lowercase because upper-case will trigger the script and it will get confused by random comments.

**Comment by johnbent** [comment]
@ChrisPerrette and @smith-371 , what do you two think about adding a requirement for additional approval before we use this autoparse? I have currently modified the import program to NOT automatically import an AUTOPARSE comment unless there are two or more 'thumbs-up' on it. For example, please check out:

![image](https://user-images.githubusercontent.com/2047294/196763523-df9b7cbf-5cbf-468e-bf94-6131ce48a53f.png)

https://github.com/tekinged/missing/issues/305#issuecomment-1284332213

Notice how it has only one thumbs-up on it? And notice the smiley face next to the thumbs-up? Right now, my script is not processing this entry because there is only one thumbs-up. I think it will improve the quality of our work if we require two thumbs-up. It will slow us down a bit but that's probably OK.

So, are you two in agreement with going a bit slower but being a bit more careful? If so, can one or both of you please go check out that autoparse comment and either make a new autoparse comment if you want to suggest changes or give it a second thumbs-up if you think it's good enough?

**Comment by johnbent** [comment]
@smith-371 and @ChrisPerrette , please review these issues and give a thumbs-up to the autoparse comment if you think it's ready:
https://github.com/tekinged/missing/issues?q=is%3Aissue+is%3Aopen+label%3A%22ready+for+import%22

**Comment by ChrisPerrette** [comment]
Out of curiosity, what does foo stand for in the autoparse arguments?

**Reply by johnbent** [reply]
'Foo' (and also 'bar') are names that programmers use when they can't think of something better. 

https://en.m.wikipedia.org/wiki/Foobar
